File: Downloads/ServiceSaloon/Service/src/app/company/services/company.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { UserStorageService } from '../../client/basic/services/storage/user-storage.service';

const BASIC_URL = "http://localhost/8050/";

@Injectable({
  providedIn: 'root'
})
export class CompanyService {

  constructor(private http: HttpClient) { }

  postAd(adDTO: any): Observable<any>{
    const userId = UserStorageService.getUserId();
    return this.http.post(BASIC_URL + `api/company/ad/${userId}`, adDTO, {
      headers : this.createAuthorizationHeader()
    })
  }

  getAllAdsByUserId(): Observable<any>{
    const userId = UserStorageService.getUserId();
    return this.http.get(BASIC_URL + `api/company/ads/${userId}`, {
      headers : this.createAuthorizationHeader()
    })
  }

  getAdById(adId:any): Observable<any>{
    return this.http.get(BASIC_URL + `api/company/ad/${adId}`, {
      headers : this.createAuthorizationHeader()
    })
  }

  updatedAd(adId:any, adDTO:any): Observable<any>{
    return this.http.put(BASIC_URL + `api/company/ad/${adId}`, adDTO,{
      headers : this.createAuthorizationHeader()
    })
  }

  deleteAd(adId:any): Observable<any>{
    return this.http.delete(BASIC_URL + `api/company/ad/${adId}`,{
      headers : this.createAuthorizationHeader()
    })
  }

  getAllAdBookings(): Observable<any>{
    const companyId = UserStorageService.getUserId();
    return this.http.get(BASIC_URL + `api/company/bookings/${companyId}`, {
      headers : this.createAuthorizationHeader()
    })
  }

  changeBookingStatus(bookingId: number, status: string): Observable<any>{

    return this.http.get(BASIC_URL + `api/company/bookings/${bookingId}/${status}`, {
      headers : this.createAuthorizationHeader()
    })
  }


  createAuthorizationHeader(): HttpHeaders{
    let authHeaders: HttpHeaders = new HttpHeaders();
    return authHeaders.set(
      'Authorization',
      'Bearer ' + UserStorageService.getToken()
    )
  }

}

File: Downloads/ServiceSaloon/src/main/java/com/Fiap/ServiceSaloon/services/company/CompanyService.java
package com.Fiap.ServiceSaloon.services.company;

import java.io.IOException;
import java.util.List;
import java.util.stream.Collectors;

import com.Fiap.ServiceSaloon.dto.AdDTO;
import com.Fiap.ServiceSaloon.dto.ReservationDTO;
import com.Fiap.ServiceSaloon.entity.Ad;

public interface CompanyService {
	
	 boolean postAd(Long userId, AdDTO adDTO) throws IOException;
	 
	 List<AdDTO> getAllAds(Long userId);
	 
	 AdDTO getAdById(Long adId);
	 
	 boolean updatedAd(Long adId, AdDTO adDTO) throws IOException;
	 
	 boolean deleteAd(Long adId);
	 
	 List<ReservationDTO> getAllAdBokings(long companyId);
	 
	 boolean changeBookingStatus(Long bookingId, String status);
	 

}
